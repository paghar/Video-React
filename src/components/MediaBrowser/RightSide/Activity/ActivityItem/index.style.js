//@flow
import styled from 'styled-components';
import Typography from 'components/shared/Typography';
export const StyledDiv = styled.div`
  outline: none;
  border: none;
  background: none;
  display: flex;
  align-items: flex-start;
  padding: 0px 27px 0px 20px;
  margin: 30px 0px;
`;
export const StyledActivityDiv = styled.div`
  flex: 1;
  padding-left: 7px;
  position: relative;
`;
export const UserNameLabel = styled(Typography)`
  outline: none;
  display: flex;
  align-self: center;
  pointer-events: none;
`;
export const TimeLabel = styled(Typography)`
  position: absolute;
  top: 0px;
  right: 0px;
`;
export const TextLabel = styled(Typography)`
  outline: none;
  display: flex;
  align-self: center;
  pointer-events: none;
`;
