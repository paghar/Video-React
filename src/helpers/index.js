//@flow

// Converts hex color to rgba string
export function hexToRgb(hex: string, opacity: number): string {
  const bigInt = parseInt(
    typeof hex === 'string' ? hex.replace('#', '') : hex,
    16
  );
  const r = (bigInt >> 16) & 255;
  const g = (bigInt >> 8) & 255;
  const b = bigInt & 255;

  return `rgba(${[r, g, b].join(',')}, ${opacity})`;
}

// finds an element parent which has specific tag name
export const findParent = (
  element: HTMLElement | EventTarget,
  parentTag: string
): any => {
  let currentNode = element;
  while (currentNode !== null) {
    if (currentNode.tagName === parentTag) {
      return currentNode;
    }
    if (currentNode.parentNode) currentNode = currentNode.parentNode;
    else currentNode = null;
  }
  return currentNode;
};
