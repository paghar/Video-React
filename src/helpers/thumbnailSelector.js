import FixImage from 'components/MediaBrowser/AssetCard/CardThumbnails/FixImage';
import Unknown from 'components/MediaBrowser/AssetCard/CardThumbnails/Unknown';
import VideoSequences from 'components/MediaBrowser/AssetCard/CardThumbnails/VideoSequences';

export const FixImageType = 'FixImageType';
export const UnknownType = 'UnknownType';
export const VideoSequencesType = 'VideoSequenceType';

export const selectThumbnailComponent = thumbnailType => {
  switch (thumbnailType) {
    case FixImageType:
      return FixImage;
    case UnknownType:
      return Unknown;
    case VideoSequencesType:
      return VideoSequences;
    default:
      return Unknown;
  }
};
