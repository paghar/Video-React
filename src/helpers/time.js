//@flow
import type { ActivityType } from 'components/MediaBrowser/RightSide/Activity/ActivityItem';
//Get Day of (activity.createdAt)
export const getActivityTime = (activitiy: ActivityType): number => {
  return parseInt(new Date(activitiy.createdAt).getDay());
};
//Get Day of today
export const getToday = (): number => {
  return parseInt(new Date(Date.now()).getDay());
};
//minute of time gets to 2 digits
export const twoDigitMinutes = (time: number): string => {
  if (time > 0) {
    const minute = new Date(time).getMinutes();
    return `${minute < 10 ? '0' + minute : minute}`;
  }
  return '';
};
//hours of time gets to 2 digits
export const twoDigitHours = (time: number): string => {
  if (time > 0) {
    const Hours = new Date(time).getHours();
    return `${Hours < 10 ? '0' + Hours : Hours}`;
  }
  return '';
};
//turn time to Sat.oct9
export const getShortDate = (time: number): string => {
  const optionDay = { weekday: 'long' };
  const optionMonth = { month: 'long' };
  const day = new Intl.DateTimeFormat('en-US', optionDay)
    .format(time)
    .slice(0, 3);
  const month = new Intl.DateTimeFormat('en-US', optionMonth)
    .format(time)
    .slice(0, 3)
    .toLowerCase();
  const date = new Date(time).getDate();
  return `${day}.${month} ${date}`;
};
//turn time to 20th-FEB-22:22 or 2017,FEB 20
export const getListViewDate = (time: number): string => {
  const year = new Date(time).getFullYear();
  const thisYear = new Date(Date.now()).getFullYear();
  const date = new Date(time).getDate();
  const optionMonth = { month: 'long' };
  const month = new Intl.DateTimeFormat('en-US', optionMonth)
    .format(time)
    .slice(0, 3)
    .toUpperCase();

  const minutes = twoDigitMinutes(time);
  const hours = twoDigitHours(time);
  if (year === thisYear) return `${date}th-${month}-${hours}:${minutes}`;
  else return `${year},${month} ${date}`;
};
