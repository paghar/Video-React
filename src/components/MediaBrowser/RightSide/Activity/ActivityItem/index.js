//@flow
import React from 'react';
import {
  StyledDiv,
  UserNameLabel,
  TimeLabel,
  TextLabel,
  StyledActivityDiv,
} from './index.style';
import Avatar from 'components/shared/Avatar';
import { Display2, Display4, Display5 } from 'theme/fonts';
import { Generals } from 'theme/colors';
import { twoDigitMinutes } from 'helpers/time';

export type ActivityType = {
  avatarSrc: string,
  name: string,
  shortName: string,
  createdAt: number,
  text: string,
  avatarColor: string,
};
type Props = {
  className?: string,
};
const ActivityItem = ({
  avatarSrc,
  name,
  shortName,
  createdAt,
  text,
  avatarColor,
  className,
}: Props & ActivityType) => {
  return (
    <StyledDiv>
      <Avatar
        backgroundColor={avatarColor}
        imgUrl={avatarSrc}
        width="24px"
        height="24px"
        borderRadius="100%"
        font={Display4}
        color={Generals.viii}
      >
        {shortName}
      </Avatar>
      <StyledActivityDiv>
        <UserNameLabel font={Display2} color={Generals.vii}>
          {name}
        </UserNameLabel>
        <TextLabel font={Display4} color={Generals.vi}>
          {text}
        </TextLabel>
        <TimeLabel font={Display5} color={Generals.vi}>
          {`${twoDigitMinutes(createdAt) &&
            new Date(createdAt).getHours()}:${twoDigitMinutes(createdAt)}`}
        </TimeLabel>
      </StyledActivityDiv>
    </StyledDiv>
  );
};
export default ActivityItem;
