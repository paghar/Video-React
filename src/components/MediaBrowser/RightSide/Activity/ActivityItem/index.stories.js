//@flow
import React from 'react';
import ActivityItem from '.';
import image from 'assets/image/christopher-campbell-rDEOVtE7vOs-unsplash.png';
export default { title: 'Activity' };
const item = {
  avatarSrc: image,
  name: 'Sara Parker',
  shortName: 'SP',
  createdAt: 1570700342376,
  text: 'Uploaded "aportman family"',
  avatarColor: '#fff',
  avatarBackgroundColor: '#5c4dff',
};
export const ActivityMenu = () => (
  <ActivityItem
    avatarSrc={item.avatarSrc}
    name={item.name}
    createdAt={item.createdAt}
    text={item.text}
    avatarColor={item.avatarColor}
    shortName={item.shortName}
  />
);
