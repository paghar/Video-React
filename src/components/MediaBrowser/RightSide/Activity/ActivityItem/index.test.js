//@flow
import React from 'react';
import ActivityItem from '.';
import renderer from 'react-test-renderer';
import { create } from 'react-test-renderer';
import { twoDigitMinutes } from 'helpers/time';
import image from 'assets/image/christopher-campbell-rDEOVtE7vOs-unsplash.png';
const item = {
  avatarSrc: image,
  shortName: 'SP',
  name: 'Sara Parker',
  createdAt: 1570700342376,
  text: 'Uploaded "aportman family"',
  avatarColor: '#fff',
};

describe('<Activity />', () => {
  it('checks the UI', () => {
    const wrapper = renderer
      .create(
        <ActivityItem
          shortName={item.shortName}
          avatarSrc={item.avatarSrc}
          name={item.name}
          createdAt={item.createdAt}
          text={item.text}
          avatarColor={item.avatarColor}
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('test the epoch input function', () => {
    expect(twoDigitMinutes(1234567890)).toBe('26');
  });
  it('test the epoch input to be 2digits', () => {
    expect(twoDigitMinutes(1570700342376)).toBe('09');
  });
});
