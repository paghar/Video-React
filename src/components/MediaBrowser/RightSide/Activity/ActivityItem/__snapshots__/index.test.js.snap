// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Activity /> checks the UI 1`] = `
.c3 {
  pointer-events: auto;
}

.c0 {
  outline: none;
  border: none;
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 0px 27px 0px 20px;
  margin: 30px 0px;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding-left: 7px;
  position: relative;
}

.c4 {
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  pointer-events: none;
}

.c6 {
  position: absolute;
  top: 0px;
  right: 0px;
}

.c5 {
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  pointer-events: none;
}

.c1 {
  background-color: #fff;
  background-image: url(christopher-campbell-rDEOVtE7vOs-unsplash.png);
  background-size: 100% 100%;
  width: 24px;
  height: 24px;
  border-radius: 100%;
  border: 0px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

<div
  className="c0"
>
  <button
    className="c1"
    height="24px"
    width="24px"
  />
  <div
    className="c2"
  >
    <span
      className="c3 c4"
    >
      Sara Parker
    </span>
    <span
      className="c3 c5"
    >
      Uploaded "aportman family"
    </span>
    <span
      className="c3 c6"
    >
      13:09
    </span>
  </div>
</div>
`;
